File: PenultimateWord/PenultimateWord.cpp
// Penultimate Word solution in C++ for CodeEval.com by Steven A. Dunn

#include <fstream>
#include <iostream>
#include <iterator>
#include <sstream>
#include <string>
#include <vector>

using namespace std;

int main(int argc, char* argv[])
{
  	ifstream inputFile(argv[1]);
	string line;

  	if(inputFile)
  	{
		while (getline(inputFile, line))
		{
			vector<string> words;
			stringstream ss(line);
		    copy(istream_iterator<string>(ss), 
		      istream_iterator<string>(), 
		      back_inserter<vector<string> >(words));
		
			cout << words[words.size() - 2] << endl;
		}
	}
}

File: PenultimateWord/c#/PenultimateWord.cs
// penultimate word soln in c# for code eval by steven a dunn

using System;
using System.IO;

class PenultimateWord {
  public static void Main(string[] args) {
    using (StreamReader reader = new StreamReader(args[0])) {
      while (!reader.EndOfStream) {
        string line = reader.ReadLine();
        if (null == line)
          continue;
        
        string[] words = line.Split(' ');
        Console.WriteLine(words[words.Length - 2]);
      }
    }
  }

}